### DISCUSSION TITLE
服务器如何主动通知客户端

### DISCUSSION THREAD

**Discussion by ericzhanchina** [cloudwu/skynet#1885] [General]
现在用的skynet提供的请求和响应的模型，做功能很清晰，比较舒服。想请教一下，对于这个socket链接，我如何能服务器主动的通知客户端数据的变化呢，现在基本都是客户端发送请求过来，我修改之后，再返回给客户端。

**Comment by CoolDesert** [comment]
客户端发请求，如果数据没有变化，服务器暂存请求，等待数据变化之后，再回应。

**Reply by ericzhanchina** [reply]
这是skynet里面的response模式吗？

**Comment by fanyh** [comment]
直接用socket.write对fd写就行了；rpc也是基于socket.write的

**Reply by ericzhanchina** [reply]
我看了一下代码在gated服务里面是持有了MsgServer模块，是否我需要修改一下MsgServer，然后如果向客户端推送数据的话，就是发给MsgServer，然后MsgServer通过socketdriver写数据给客户端？

**Comment by ericzhanchina** [comment]
@cloudwu 云风大佬可以给指导一下吗，谢谢
